### PR TITLE
Update to vertx 4.4.2

### LINKED ISSUE #33468: Upgrade vertx to 4.4.* to allow TLS Server certificate rotation without restarts
### Description

I'm interested in creating a Vert.x server that updates TLS certificates without restarts using the new Vert.x feature implemented in https://github.com/eclipse-vertx/vert.x/commit/f288f356bcdd9a03273cd1bb9c46b9a720d604c6 and documented at https://vertx.io/docs/vertx-core/java/#_updating_ssltls_configuration

The feature is available with Vertx 4.4 but the current Vertx version for Quarkus is 4.3.

### Implementation ideas

Upgrade vertx to 4.4.2

### PR DESCRIPTION
**Pull request by cescoffier**
The first commit is the actual update. 
The second one fixes an OpenTelemetry test that was not testing the real thing. 

Fixes: #33468

### CODE REVIEW

File: integration-tests/opentelemetry-vertx/src/test/java/io/quarkus/it/opentelemetry/vertx/SqlClientTest.java [hunk 1]
```diff
@@ -86,25 +56,35 @@ void sqlClient() {
...
+
+        inMemorySpanExporter.getFinishedSpanItems().forEach(sd -> {
+            System.out.println(sd);
+        });
+
+        // We cannot rely on the order, we must identify the spans.
+        SpanData httpSpan = inMemorySpanExporter.getFinishedSpanItems().stream()
+                .filter(sd -> sd.getName().contains("GET /sqlClient")).findFirst().orElseThrow();
+        SpanData querySpan = inMemorySpanExporter.getFinishedSpanItems().stream()
+                .filter(sd -> sd.getName().contains("SELECT USERS")).findFirst().orElseThrow();
```
Comment (brunobat) [inline integration-tests/opentelemetry-vertx/src/test/java/io/quarkus/it/opentelemetry/vertx/SqlClientTest.java:74]: There is a 3rd span and it's not asserted. What is it?

Comment (cescoffier) [reply integration-tests/opentelemetry-vertx/src/test/java/io/quarkus/it/opentelemetry/vertx/SqlClientTest.java:74]: It's the span for the table creation. Not sure we need to assert it. There are no relations with the other spans (maybe that's what we need to check).

Comment (brunobat) [reply integration-tests/opentelemetry-vertx/src/test/java/io/quarkus/it/opentelemetry/vertx/SqlClientTest.java:74]: makes sense

Comment (cescoffier) [reply integration-tests/opentelemetry-vertx/src/test/java/io/quarkus/it/opentelemetry/vertx/SqlClientTest.java:73]: Done.

**Comment by geoand** [comment]
@mkouba I saw `PausedSchedulerTest` failing in another PR as well.

**Comment by mkouba** [comment]
>  I saw PausedSchedulerTest failing in another PR as well.

Hm, that's weird. It should not be a timinig issue. 

@geoand Was it also the JVM Tests - JDK 17 Windows job that failed in the other PR?

**Comment by geoand** [comment]
I don't remember honestly...

**Comment by mkouba** [comment]
> I don't remember honestly...

Ok, let's watch this test closely. TBH I don't see a reason why this should fail and obviously it does not fail locally...

**Comment by Sanne** [comment]
nevermind, I was able to do it.

**Review by Sanne — APPROVED** [review]

**Pull request merged**